edc-extensions/validators/empty-asset-selector/src/main/java/org/eclipse/tractusx/edc/validators/emptyassetselector/EmptyAssetSelectorValidator.java

```diff
@@ -0,0 +1,43 @@
+/********************************************************************************
+ * Copyright (c) 2025 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.edc.validators.emptyassetselector;
+
+import org.eclipse.edc.spi.query.CriterionOperatorRegistry;
+import org.eclipse.edc.validator.jsonobject.JsonObjectValidator;
+import org.eclipse.edc.validator.jsonobject.validators.MandatoryValue;
+import org.eclipse.edc.validator.jsonobject.validators.OptionalIdNotBlank;
+import org.eclipse.edc.validator.jsonobject.validators.model.CriterionValidator;
+
+import static org.eclipse.edc.connector.controlplane.contract.spi.types.offer.ContractDefinition.CONTRACT_DEFINITION_ACCESSPOLICY_ID;
+import static org.eclipse.edc.connector.controlplane.contract.spi.types.offer.ContractDefinition.CONTRACT_DEFINITION_ASSETS_SELECTOR;
+import static org.eclipse.edc.connector.controlplane.contract.spi.types.offer.ContractDefinition.CONTRACT_DEFINITION_CONTRACTPOLICY_ID;
+
+public class EmptyAssetSelectorValidator {
+
+    public static JsonObjectValidator instance(CriterionOperatorRegistry criterionOperatorRegistry) {
+        return JsonObjectValidator.newValidator()
```

Comment:
this is pretty much duplicated from upstream and it makes me think that maybe there could be a better way to tackle that, like have the possibility to provide additional validators to the already existent validators through composition. Not for this PR, but this is maybe a topic that could be raised upstream.